tests/suite/reconfig_test.go

```diff
@@ -0,0 +1,638 @@
+		Expect(err).ToNot(HaveOccurred())
+
+		eventsCount, err := framework.GetEventsCount(promInstance, ngfPodName)
+		Expect(err).ToNot(HaveOccurred())
+
+		eventsAvgTime, err := framework.GetEventsAvgTime(promInstance, ngfPodName)
+		Expect(err).ToNot(HaveOccurred())
+
+		eventsBuckets, err := framework.GetEventsBuckets(promInstance, ngfPodName)
+		Expect(err).ToNot(HaveOccurred())
+
+		logs, err := resourceManager.GetPodLogs(ngfNamespace, ngfPodName, &core.PodLogOptions{
+			Container: "nginx-gateway",
+		})
+		Expect(err).ToNot(HaveOccurred())
+
+		timeToReadyTotal, err := calculateTimeToReadyTotal(logs, timeToReadyStartingLogSubstring)
+		Expect(err).ToNot(HaveOccurred())
+
+		timeToReadyAvgSingle, err := calculateTimeToReadyAverage(logs)
+		Expect(err).ToNot(HaveOccurred())
+
+		results := reconfigTestResults{
+			TestDescription:      testDescription,
+			EventsBuckets:        eventsBuckets,
+			ReloadBuckets:        reloadBuckets,
+			NumResources:         resourceCount,
+			TimeToReadyTotal:     timeToReadyTotal,
+			TimeToReadyAvgSingle: timeToReadyAvgSingle,
+			NGINXReloads:         int(reloadCount),
+			NGINXReloadAvgTime:   int(reloadAvgTime),
+			EventsCount:          int(eventsCount),
+			EventsAvgTime:        int(eventsAvgTime),
+		}
+
+		err = writeReconfigResults(outFile, results)
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	When("resources exist before startup", func() {
+		testDescription := "Test 1: Resources exist before startup"
+
+		It("gathers metrics after creating 30 resources", func() {
+			resourceCount := 30
+			timeToReadyStartingLogSubstring := "Starting NGINX Gateway Fabric"
+			test := createResourcesGWLast(resourceCount)
+
+			Expect(test).To(Succeed())
+			Expect(checkResourceCreation(resourceCount)).To(Succeed())
+
+			ngfPodName, startTime := deployNGFReturnsNGFPodNameAndStartTime()
+
+			collectMetrics(
+				testDescription,
+				resourceCount,
+				timeToReadyStartingLogSubstring,
+				ngfPodName,
+				startTime,
+			)
+		})
+
+		It("gathers metrics after creating 150 resources", func() {
+			resourceCount := 150
+			timeToReadyStartingLogSubstring := "Starting NGINX Gateway Fabric"
+			test := createResourcesGWLast(resourceCount)
+
+			Expect(test).To(Succeed())
+			Expect(checkResourceCreation(resourceCount)).To(Succeed())
+
+			ngfPodName, startTime := deployNGFReturnsNGFPodNameAndStartTime()
+
+			collectMetrics(
+				testDescription,
+				resourceCount,
+				timeToReadyStartingLogSubstring,
+				ngfPodName,
+				startTime,
+			)
+		})
+	})
+
+	When("NGF and Gateway resource are deployed first", func() {
+		testDescription := "Test 2: Start NGF, deploy Gateway, create many resources attached to GW"
+
+		It("gathers metrics after creating 30 resources", func() {
+			resourceCount := 30
+			timeToReadyStartingLogSubstring := "Reconciling the resource\",\"controller\":\"httproute\""
+			test := createResourcesRoutesLast(resourceCount)
+
+			ngfPodName, startTime := deployNGFReturnsNGFPodNameAndStartTime()
```

Comment:
won't this still deploy routes/apps before NGF exists?

Reply:
I don't think so? 

`Expect(test).To(Succeed())` will deploy the routes/apps and in 

`deployNGFReturnsNGFPodNameAndStartTime()` it has

```
setup(getDefaultSetupCfg())
podNames, err := framework.GetReadyNGFPodNames(k8sClient, ngfNamespace, releaseName, timeoutConfig.GetTimeout)
Expect(err).ToNot(HaveOccurred())
Expect(podNames).To(HaveLen(1))
```
which should wait for NGF to deploy no?

Reply:
Oh you're right, I misread when the resources were actually being created.

Reply:
Actually I think you may have been right as I think `test := createResourcesRoutesLast(resourceCount)` actually calls the function instead of assigning test to the function.

Reply:
Yeah, so you should just be able to fix it so the function is called when needed. No sense in storing in a variable.